### DISCUSSION TITLE
Including non-screen files in directory

### DISCUSSION THREAD

**Discussion by wjaykim** [expo/router#318] [Ideas]
Next.js supports [customizing the extensions for pages](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory), which enables developers to include non-page files under `pages` directory.

So what I want for expo-router is like this:

- apps
  - foo
    - index.screen.tsx
    - splitted-component.tsx
  - bar.screen.tsx
  - bar.test.ts
  
Can I expect this feature to be implemented in future?

**Comment by mauricedoepke** [comment]
If I got you right there are some discussions regarding this in the following thread:
https://github.com/expo/router/discussions/1

The keyword to search for is "colocate" components.

Seems like there is no official way or plan for this yet.
